### ISSUE TITLE
Duplicate IDs for link targets (duplicate-id-active)

### ISSUE THREAD

**Issue by aardrian** [dequelabs/axe-core#3809]
### Product

axe-core

### Product Version

4.5

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
Not flag duplicate IDs on non-interactive nodes under the rule [duplicate-id-active](https://dequeuniversity.com/rules/axe/4.5/duplicate-id-active).

#### Actual
Flags duplicate IDs on non-interactive nodes under the rule [duplicate-id-active](https://dequeuniversity.com/rules/axe/4.5/duplicate-id-active).

#### How to Reproduce
- [Duplicate IDs in Link Targets](https://cdpn.io/pen/debug/vYrQEbr) on CodePen, debug mode.
- [Duplicate IDs in Link Targets, editable](https://codepen.io/aardrian/pen/vYrQEbr) on CodePen.

This element is not interactive (it has no implicit nor explicit interactive role):
```
<h2 id="Sandwich" tabindex="-1">Second Sandwich</h2>
```

It has `tabindex="-1"`, so I suspect axe is treating it as de facto interactive. Especially since the following is flagged under rule [duplicate-id](https://dequeuniversity.com/rules/axe/4.4/duplicate-id) instead:
```
<h2 id="Hammer">Second Hammer</h2>
```

The issue I am raising is not its interactivity, though. The issue I am raising is that this affects all users and therefore is outside the scope of WCAG.

#### Additional context
This Axe rule is being applied because it is considered "active", and the rule is "IDs of active elements must be unique". I think the axe rule argument is too broad:
> The value assigned to active ID attributes on focusable elements must be unique to prevent the second instance from being overlooked by assistive technology. Put another way, active ID attributes may not be used more than once on focusable elements in the same document; focusable active elements require unique IDs for assistive technology to distinguish one element from another. 

Any instance after the first will be overlooked by the _browser_ itself, thus affecting all users. Granted, as with all web page bugs it can (and will) have an outsized impact on many users with disabilities.

In addition:
1. A duplicate ID alone is already covered in rule [duplicate-id](https://dequeuniversity.com/rules/axe/4.4/duplicate-id) (assuming the check for `tabindex="-1"` is the trigger for duplicate-id-active and that exclusion can be removed from duplicate-id);
2. I am not filing a similar issue on rule [duplicate-id](https://dequeuniversity.com/rules/axe/4.4/duplicate-id) simply because if you do not agree with me here, it is almost definitely mooted there;
3. If 4.1.1 goes away, this will definitely not be a 4.1.1 issue;
4. I think this also aligns with [Dylan's suggestion for tweaking the 4.1.1 language](https://github.com/w3c/wcag/issues/2525#issuecomment-1319925864) (before it was decided to maybe purge 4.1.1):

>One consideration, to allow automated testing to assist with more efficient testing, would be to change 4.1.1 to disallow only duplicate IDs that are either:
>
> 1. referenced by some other element (this would handle all the accName issues), or
> 2. on an interactive element (this would handle all the activation of interactive elements issues)
>
> and allow duplicate IDs on static content that is not one of the two above cases

I understand the context was about rephrasing 4.1.1 and not how it is interpreted by many, but it seems to track with my suggestion.

**Comment by dylanb** [comment]
@aardrian we have decided that having a tabindex on an element that is otherwise not focusable, makes it interactive, in this instance I assume it is because some other element like a skip link is using the ID to target the focus, this means that the focus could go to the wrong element and this is going to affect users, dependent on the keyboard more than other users. The value provided to the developer here is higher than the potential for a mis-categorization.
